addons/web/static/src/js/views/graph/graph_owl_renderer.js

```diff
@@ -0,0 +1,927 @@
+            } else if (this.props.mode === 'line') {
+                label = this._relabelling(label, dataset.originIndex);
+                if (this.props.processedGroupBy.length > 1 || this.props.origins.length > 1) {
+                    label = label + "/" + dataset.label;
+                }
+                value = this._formatValue(item.yLabel);
+                boxColor = dataset.borderColor;
+            } else {
+                if (label.isNoData) {
+                    value = this._formatValue(0);
+                } else {
+                    value = this._formatValue(dataset.data[item.index]);
+                }
+                label = this._relabelling(label, dataset.originIndex);
+                if (this.props.origins.length > 1) {
+                    label = dataset.label + "/" + label;
+                }
+                boxColor = dataset.backgroundColor[item.index];
+            }
+            return {
+                label: label,
+                value: value,
+                boxColor: boxColor,
+            };
+        }
+
+        /**
+         * This function extracts the information from the data points in tooltipModel.dataPoints
+         * (corresponding to datapoints over a given label determined by the mouse position)
+         * that will be displayed in a custom tooltip.
+         *
+         * @private
+         * @param {Object} tooltipModel see chartjs documentation
+         * @return {Object[]}
+         */
+        _getTooltipItems(tooltipModel) {
+            var self = this;
+            var data = this.chart.config.data;
+
+            var orderedItems = tooltipModel.dataPoints.sort(function (dPt1, dPt2) {
+                return dPt2.yLabel - dPt1.yLabel;
+            });
+            return orderedItems.reduce(
+                function (acc, item) {
+                    acc.push(self._getTooltipItemContent(item, data));
+                    return acc;
+                },
+                []
+            );
+        }
+
+        /**
+         * Determine how to relabel a label according to a given origin.
+         * The idea is that the getLabel function is in general not invertible but
+         * it is when restricted to the set of dataPoints coming from a same origin.
+
+         * @private
+         * @param {Array} label
+         * @param {Array} originIndex
+         * @returns {string}
+         */
+        _relabelling(label, originIndex) {
+            if (label.isNoData) {
+                return label[0];
+            }
+            var i = this.props.comparisonFieldIndex;
+            if (_.contains(['bar', 'line'], this.props.mode) && i === 0) {
+                // here label is an array of length 1 and contains a number
+                return this.dateClasses.representative(label, originIndex) || '';
+            } else if (this.props.mode === 'pie' && i === 0) {
+                // here label is an array of length at least one containing string or numbers
```

Comment:
If you want to work more. You can refactor the code  after the conversion. The idea is that we should use simple labels and use metadata passed to Chart.js to remap things. This would become more readable. The fact that I used
arrays as label was good to formalise things but maybe not the best for readability

Reply:
We could also maybe use JSON.stringify like in pivot model?

Reply:
As discussed on Discord, not a necessity.